ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -198,6 +195,8 @@ public void setupResources(final Configurable<?> config, final KsqlRestConfig ap
   @Override
   public void start() throws Exception {
     super.start();
+    final KsqlConfig ksqlConfigWithPort = buildConfigWithPort();
+    configurables.forEach(c -> c.configure(ksqlConfigWithPort));
```

Comment:
Once the Jetty server has been started we know the actual ports, so can now build more complete config and configure those components that are lazily configured.